mad-dog/src/tests/test_userReplicaSetNodes/index.js

```diff
@@ -0,0 +1,58 @@
+const { addAndUpgradeUsers } = require('../../helpers.js')
+const verifyValidCNs = require('./verifyValidCN')
+const deregisterCreatorNode = require('./deregisterCreatorNode.js')
+const stopRandomCreatorNode = require('./stopRandomCreatorNode.js')
+const setNumCreatorNodes = require('./setNumCreatorNodes.js')
+
+/**
+ * Tests that the user replica sets update when a node is deregistered or down
+ */
+const userReplicaSetNodes = async ({
+  numUsers = 10,
+  numCreatorNodes = 10,
+  iterations,
+  executeAll,
+  executeOne
+}) => {
+
+  // Make sure there are 4 creator nodes
+  let creatorNodeIDToInfoMapping = {}
+
+  let walletIndexToUserIdMap = {}
+  // Create users
+  // Initialize users
+  try {
+    walletIndexToUserIdMap = await addAndUpgradeUsers(
+      numUsers,
+      executeAll,
+      executeOne
+    )
+  } catch (e) {
+    return { error: `Issue with creating and upgrading users: ${e}` }
+  }
+
+
+  for (let iteration = 0; iteration < iterations; iteration++) {
+    // Ensure that there are 10 registered CN before each iteration
+    creatorNodeIDToInfoMapping = await setNumCreatorNodes(numCreatorNodes, executeOne)
+
+    const deregisteredCreatorNodeId = await deregisterCreatorNode(creatorNodeIDToInfoMapping)
+
+    await new Promise(resolve => setTimeout(resolve, 10 * 1000))
```

Comment:
surprised 10s has been enough for all content to adjust locally, even with 10 users

Reply:
Oh well I didn't know what to set this to, figured you'd change it when running tests later.
It is timed based right now, but we could also do something like block number or use some other waiting heuristic if you wanted to.
None of the tests pass right now 🙃

Reply:
hahaha right, forgot that nothing works, yep sounds good. we'll adjust it next week, aiming to get the state machine working by end of week